src/media-groups.js

```diff
@@ -0,0 +1,756 @@
+    segmentLoader.on('error', onError[type](type, settings));
+  },
+  /**
+   * Setup TextTracks for the closed-caption groups
+   *
+   * @param {String} type
+   *        MediaGroup type
+   * @param {Object} settings
+   *        Object containing required information for media groups
+   * @function initialize['CLOSED-CAPTIONS']
+   */
+  'CLOSED-CAPTIONS': (type, settings) => {
+    const {
+      tech,
+      master: { mediaGroups: masterGroups},
+      mediaGroups: {
+        [type]: {
+          groups,
+          tracks
+        }
+      }
+    } = settings;
+
+    for (let masterGroup in masterGroups[type]) {
+      if (!groups[masterGroup]) {
+        groups[masterGroup] = [];
+      }
+
+      for (let label in masterGroups[type][masterGroup]) {
+        let properties = masterGroups[type][masterGroup][label];
+
+        // We only support CEA608 captions for now, so ignore anything that
+        // doesn't use a CCx INSTREAM-ID
+        if (!properties.instreamId.match(/CC\d/)) {
+          continue;
+        }
+
+        // No PlaylistLoader is required for Closed-Captions because the captions are
+        // embedded within the video stream
+        groups[masterGroup].push(videojs.mergeOptions({ id: label }, properties));
+
+        if (typeof tracks[label] === 'undefined') {
+          const track = tech.addRemoteTextTrack({
+            id: properties.instreamId,
+            kind: 'captions',
+            enabled: false,
+            language: properties.language,
+            label
+          }, false).track;
+
+          tracks[label] = track;
+        }
+      }
+    }
+  }
+};
+
+/**
+ * Returns a function used to get the active group of type provided
+ *
+ * @param {String} type
+ *        MediaGroup type
+ * @param {Object} settings
+ *        Object containing required information for media groups
+ * @return {Function}
+ *         Function that returns the active media group for the provided type. Takes an
+ *         optional paramter {TextTrack} track. If no track is provided, a list of all
+ *         variants in the group, otherwise the variant corresponding to the provided
+ *         track is returned.
+ * @function activeGroup
+ */
+export const activeGroup = (type, settings) => (track) => {
+  const {
+    masterPlaylistLoader,
+    mediaGroups: { [type]: { groups } }
+  } = settings;
+
+  const media = masterPlaylistLoader.media();
+
+  if (!media) {
+    return null;
+  }
+
+  let result = null;
+
+  if (media.attributes[type]) {
+    result = groups[media.attributes[type]];
+  }
+
+  result = result || groups.main;
+
+  if (typeof track === 'undefined') {
+    return result;
+  }
+
+  if (track === null) {
+    // An active track was specified so a corresponding group is expected. track === null
+    // means no track is currently active so there is no corresponding group
+    return null;
+  }
+
+  return result.reduce((final, props) => props.id === track.id ? props : final, null);
```

Comment:
Preference, but might be clearer as `result.filter((props) => props.id === track.id)[0] || null`